tests/sceneManager_test.cpp

```diff
@@ -10,7 +10,53 @@ using generators::PointSource;
 using objects::Object;
 using objects::TriangleObj;
 
-const double kSkipFreq = 1000;
 generators::FakeRandomGen fakeRand;
 
-// TODO: Change all doubles to float
+class SceneManagerTest : public ::testing::Test
+{
+protected:
+    bool performHitAtEnergyCollectors(const Ray &straightUp, RayHitData *hitData)
+    {
+        for (objects::EnergyCollector *energyCol : manager.getEnergyCollectors())
+        {
+            const double kSkipFreq = 1000;
+            if (energyCol->hitObject(straightUp, kSkipFreq, hitData))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    SceneManager manager;
+};
+
+TEST_F(SceneManagerTest, EnergyCollectorPositionsCheck)
+{
+    for (const auto &collectorPtr : manager.getEnergyCollectors())
+    {
+        std::cout << *collectorPtr << std::endl;
+    }
+
+    ASSERT_EQ(manager.getEnergyCollectors().size(), constants::kPopulation) << "wrong numebr of energy collectors";
+
+    RayHitData hitData;
+    Ray straightUp(core::Vec3(0, 0, 0), core::Vec3(0, 0, 1));
+
+    ASSERT_TRUE(performHitAtEnergyCollectors(straightUp, &hitData)) << "no hit: " << straightUp;
+
+    core::Vec3 referenceCollisionPoint(0, 0, manager.simulatiorRadius() / 2 - manager.collectorRadius());
+    ASSERT_EQ(hitData.collisionPoint, referenceCollisionPoint) << "Invalid hit point from: " << straightUp;
+
+    Ray againstXAxis(core::Vec3(0, 0, 0), core::Vec3(-1, 0, 0));
+    ASSERT_TRUE(performHitAtEnergyCollectors(againstXAxis, &hitData)) << "no hit: " << straightUp;
+
+    Vec3 referenceCollisionPoint2(-(manager.simulatiorRadius() / 2 - manager.collectorRadius()), 0, 0);
+    ASSERT_EQ(hitData.collisionPoint, referenceCollisionPoint2) << "Invalid hit point from: " << straightUp;
+
+    Ray againstYAxis(core::Vec3(0, 0, 0), core::Vec3(0, 1, 0));
+    ASSERT_TRUE(performHitAtEnergyCollectors(againstYAxis, &hitData)) << "no hit: " << straightUp;
```

Comment:
Incorrect ray in the error message. This is what you get for copy-pasting code ;-)

Reply:
that was ctr + H this time hahha

Reply:
before I got tempRay1, tempRay2, tempRay3, I made a Typo, and they were all replaced evetually with straightUp haha

Reply:
done